packages/peer-discovery-mdns/src/query.ts

```diff
@@ -92,20 +103,45 @@ export function gotQuery (qry: QueryPacket, mdns: MulticastDNS, peerName: string
       data: peerName + '.' + serviceTag
     })
 
-    multiaddrs.forEach((addr) => {
-      // spec mandates multiaddr contains peer id
-      if (addr.getPeerId() != null) {
+    multiaddrs
+      // mDNS requires link-local addresses only
+      // https://github.com/libp2p/specs/blob/master/discovery/mdns.md#issues
+      .filter(isLinkLocal)
+      .forEach((addr) => {
+        const data = 'dnsaddr=' + addr.toString()
+
+        // TXT record fields have a max data length of 255 bytes
+        // see 6.1 - https://www.ietf.org/rfc/rfc6763.txt
+        if (data.length > 255) {
+          log('multiaddr %a is too long to use in mDNS query response', addr)
```

Comment:
This will likely lead to a lot of circuit relay webtransport multiaddrs failing, won't it?